app/controllers/samples_controller.rb

```diff
@@ -969,14 +969,24 @@ def results_folder
   end
 
   # GET /samples/:id/stage_results
+  # GET /samples/:id/stage_results.json
   def stage_results
     if current_user.allowed_feature_list.include?("pipeline_viz")
-      @results = {}
-      @sample.first_pipeline_run.pipeline_run_stages.each do |stage|
+      pipeline_run = @sample.first_pipeline_run
+      pipeline_version = pipeline_run.report_info_params[:pipeline_version]
```

Comment:
Add a check on pipeline_run: Lets make sure we do not crash if there is no pipeline run yet... It is not supposed to happen when you access this endpoint but better to protect the code.